### ISSUE TITLE
Move elasticsearch to own extension

### ISSUE THREAD

**Issue by DanielSiepmann** [Codappix/search_core#169]
As soon as we support multiple search engines, e.g. algolia, we will move elasticsearch to search_elasticsearch.

search_core will never be available through TER, only concrete extensions like search_algolia or search_elasticsearch.

Those just require search_core as a library / TYPO3 extension via composer. If the extension is not installed via composer, the dependencies should be bundles while releasing to TER. While we have to check the current state at the moment we want to make the shift. The reason to check again, is:

>  The short-term goal is, that this custom composer repository becomes obsolete and all extensions published to TER are also available on Packagist. 
> https://typo3.org/project/news/this-month-in-typo3/this-month-in-typo3-october-2018-issue-7/

**Comment by DanielSiepmann** [comment]
To quote Benjamin:

> How I would probably solve this is like this;
> - search_core (base extension, not released in TER)
> - search_elastic (composer requirement for search_core, but included as .phar when not via composer)
> - search_algolia (composer requirement for search_core, but included as .phar when not via composer)
> - etc. :slightly_smiling_face:

**Comment by konnisoelch** [comment]
Makes sense to me, i guess the commitment of other people would also be higher, since they only need to update the core functionality and not some other search engines, which they propably don't use :)

**Comment by marvinhuebner** [comment]
I think that's a very good idea. Thus, the chore functionality can be developed independently of the respective search engines.

To the point with the ter and the .phar file. My personal opinion is that someone who deals with elasticsearch or algolia probably has the necessary knowledge to set up a typo3 project via composer. That's why I would not put much energy into publish search_elasticsearch or search_algolia in ter, instead these extensions composer only and can be installed via packagist . Sooner or later, the ter should be replaced anyway by packagist (which I think I have at least heard).
